File: src/main/java/sprykerFileCreator/model/definition/DefinitionProvider.java
package pav.sprykerFileCreator.model.definition;

import pav.sprykerFileCreator.model.definition.spryker.*;

import java.util.HashMap;

public class DefinitionProvider implements DefinitionProviderInterface {

    public static HashMap<String, ClassDefinitionInterface> classDefinitions = new HashMap<>();

    public DefinitionProvider() {
        this.addClassDefinition(new ClientDefinition());
        this.addClassDefinition(new ClientFactoryDefinition());
        this.addClassDefinition(new ClientDependencyProviderDefinition());
        this.addClassDefinition(new ClientPluginDefinition());
        this.addClassDefinition(new YvesControllerDefinition());
        this.addClassDefinition(new YvesFactoryDefinition());
        this.addClassDefinition(new YvesDependencyProviderDefinition());
        this.addClassDefinition(new YvesPluginDefinition());
        this.addClassDefinition(new ZedControllerDefinition());
        this.addClassDefinition(new ZedQueryContainerDefinition());
        this.addClassDefinition(new ZedFacadeDefinition());
        this.addClassDefinition(new ZedBusinessFactoryDefinition());
        this.addClassDefinition(new ZedPersistenceFactoryDefinition());
        this.addClassDefinition(new ZedCommunicationFactoryDefinition());
        this.addClassDefinition(new ZedConfigDefinition());
        this.addClassDefinition(new ZedPluginDefinition());
        this.addClassDefinition(new ZedDependencyProviderDefinition());
    }

    public ClassDefinitionInterface getDefinitionByType(String classType) throws Exception {

        ClassDefinitionInterface classDefinition = classDefinitions.get(classType);

        if (classDefinition == null) {
            throw new Exception("Class definition was not found: " + classType);
        }

        return classDefinition;
    }

    public HashMap<String, ClassDefinitionInterface> getAllClassDefinitions() {
        return classDefinitions;
    }

    private void addClassDefinition(ClassDefinitionInterface classDefinition) {
        classDefinitions.put(classDefinition.getClassType(), classDefinition);
    }
}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/DefinitionProvider.kt
package com.github.tobi812.sprykerplugin.models.definitions

import com.github.tobi812.sprykerplugin.models.definitions.spryker.*
import java.lang.Exception
import java.util.HashMap

class DefinitionProvider : DefinitionProviderInterface {

    companion object {
        var classDefinitions = HashMap<String, ClassDefinitionInterface>()
    }

    init {
       this.addClassDefinition(ClientDefinition())
       this.addClassDefinition(ClientFactoryDefinition())
       this.addClassDefinition(ClientDependencyProviderDefinition())
       this.addClassDefinition(ClientPluginDefinition())
       this.addClassDefinition(YvesControllerDefinition())
       this.addClassDefinition(YvesFactoryDefinition())
       this.addClassDefinition(YvesDependencyProviderDefinition())
       this.addClassDefinition(YvesPluginDefinition())
       this.addClassDefinition(ZedControllerDefinition())
       this.addClassDefinition(ZedQueryContainerDefinition())
       this.addClassDefinition(ZedFacadeDefinition())
       this.addClassDefinition(ZedBusinessFactoryDefinition())
       this.addClassDefinition(ZedPersistenceFactoryDefinition())
       this.addClassDefinition(ZedCommunicationFactoryDefinition())
       this.addClassDefinition(ZedConfigDefinition())
       this.addClassDefinition(ZedPluginDefinition())
       this.addClassDefinition(ZedDependencyProviderDefinition())
    }

    @Throws(Exception::class)
    override fun getDefinitionByType(classType: String): ClassDefinitionInterface {
        if (!classDefinitions.containsKey(classType)) {
            throw Exception("Class definition was not found: $classType")
        }

        return classDefinitions[classType]!!
    }

    override val allClassDefinitions: HashMap<String, ClassDefinitionInterface> =
        classDefinitions

    private fun addClassDefinition(classDefinition: ClassDefinitionInterface) {
        classDefinitions[classDefinition.classType] = classDefinition
    }
}